enhancements/ota/distribute-secondary-metadata-as-container-image.md

```diff
@@ -0,0 +1,165 @@
+---
+title: cincinnati-graph-data-package-distribute-and-consume-in-a-container-image-repository
+authors:
+  - "@sjunker"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2020-05-07
+last-updated: 2020-05-07
+status: provision
+see-also:
+-
+replaces: []
+superseded-by: []
+---
+
+# cincinnati-graph-data: package, distribute, and consume in a container image repository
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [x] Graduation criteria
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions
+
+### Cincinnati Git(Hub) scraper deprecation
+We need to decide whether we want to maintain the Git(Hub) scraper plugin in the future or drop it altogether.
+
+
+### Content trust
+Trust in the container image registry and in the container image content is possibly an issue if we an approach where Cincinnati uses a mutable tag, e.g. `latest` tag to receive the most recent content from the container registry.
+Assuming we're using Quay as an image registry, [which does not support Docker Content Trust](https://support.coreos.com/hc/en-us/articles/115000289114-Docker-Content-Trust-support).
+If we decide that to ensure the image integrity, which means do we have which do not further complicate the workflow?
+
+### Product name for the CPaaS integration
+Which product name should this image be released under?
+Related to this is that it's currently unclear if the Cincinnati image itself will be published under its own product name or now.
+Further note: the repository for the graph-data images will not be semantically versioned.
```

Comment:
The graph-data format is likely to evolve.  Now that Cincinnati is consuming the format directly, we should add a version identifier to the repository so Cincinnati can fail fast on snapshots it does not understand.  But that's independent of this enhancement, which is just about how we get the content from the graph-data repo over to Cincinnati.

Reply:
Proposal for semantically versioning the graph-data openshift/cincinnati-graph-data#233 ;).

Reply:
I intended to write about the release information rather than its data format. I do agree that versioning the data format, and possibly versioning the repository according to the data format changes would make sense to inform consumers of breaking changes. This assumes that the data itself will foreseeably never have to be versioned.